src/main/java/com/mixpanel/android/util/RemoteService.java

```diff
@@ -10,7 +10,7 @@
 
 
 public interface RemoteService {
-    boolean isOnline(Context context);
+    boolean isOnline(Context context, OfflineMode offlineMode);
```

Comment:
Hi @patedit thanks for the comments !  
I agree with you that adding `OfflineMode` on a call to `isOnline(Context)` is odd. Specially when you have it available on `MPConfig` that is a Singleton. But it follows the same pattern used for `SSLSocketFactory` on `byte[] performRequest(String , Map<String, Object> , SSLSocketFactory)` and was why I choose this way.